dashboard-frontend/src/pages/StudyPage/useStudy.tsx

```diff
@@ -0,0 +1,96 @@
+import { useMemo } from 'react';
+import { useQuery, queryCache } from 'react-query';
+import Chance from 'chance';
+import { lastValue } from '../../helpers/arrays';
+import { computeStudyProgressDataFrom } from '../../helpers/study';
+import useAuthenticatedApi from '../../hooks/useAuthenticatedApi';
+import {
+  StudyProgressResource,
+  StudySegmentProgressResource,
+} from '../../types/study';
+
+const chance = Chance();
+
+const useStudy = (slug: string) => {
+  const studyQuery = useStudyQuery(slug);
+  const studySegmentsProgressQuery = useStudySegmentsProgressQuery(slug);
+
+  const segmentsProgressOverTime = useMemo(
+    () => studySegmentsProgressQuery.data?.data ?? [],
+    [studySegmentsProgressQuery.data]
+  );
+
+  const progressOverTime: StudyProgressResource[] = useMemo(() => {
+    if (!segmentsProgressOverTime.length) {
+      return [getStudyDefaultProgress()];
+    }
+
+    return segmentsProgressOverTime.map(computeStudyProgress);
+  }, [segmentsProgressOverTime]);
+
+  const currentSegmentsProgress = lastValue(segmentsProgressOverTime);
+
+  const currentProgress: StudyProgressResource = useMemo(
+    () => computeStudyProgress(currentSegmentsProgress),
+    [currentSegmentsProgress]
+  );
+
+  const isLoading = !studyQuery.data || !studySegmentsProgressQuery.data?.data;
+
+  const anyErrorDuringLoading =
+    isLoading && (studyQuery.isError || studySegmentsProgressQuery.isError);
+
+  return {
```

Comment:
I can't see this type anywhere, but it seems like a pretty important thing to have as a type, as you're using it everywhere, no? this "study" object?

Reply:
The type is inferred, no need to type it explicitly 🎉  (Typescript is a structural type system).